internal/config/config.go

```diff
@@ -310,6 +320,41 @@ func setClowderConfiguration(v *viper.Viper, clowderConfig *clowder.AppConfig) {
 	// Metrics configuration
 	v.Set("metrics.path", clowderConfig.MetricsPath)
 	v.Set("metrics.port", clowderConfig.MetricsPort)
+
+	// Set TLS configuration
+	processPublicEndpoint("rbac", "v1", v, clowderConfig)
+}
+
+func processPublicEndpoint(serviceName, version string, v *viper.Viper, clowderConfig *clowder.AppConfig) {
+	if serviceName == "" {
+		panic("'serviceName' is an empty string")
+	}
+	if version == "" {
+		panic("'version' is an empty string")
+	}
+	if v == nil {
+		panic("'v' is nil")
+	}
+	if clowderConfig == nil {
+		panic("'clowderConfig' is nil")
+	}
+	if clowderConfig.TlsCAPath != nil {
+		paramPath := "clients." + serviceName + "_base_url"
+		serviceEndpoint := getPublicEndpoint(serviceName, clowderConfig)
+		if serviceEndpoint != nil {
+			serviceBaseURLString := "https://" + serviceEndpoint.Hostname
```

Comment:
The protocol should be `http` if TlsPort is not defined.   https://gitlab.cee.redhat.com/insights-platform/advisor-backend/-/merge_requests/490/diffs#652c5e5dd60867263fdb20ba5922d3d34a615885_102_103

Reply:
thanks! updated with the current state!